node/pkg/driver/node.go

```diff
@@ -502,6 +508,14 @@ func (d *NodeService) publishRawBlockVolume(mpathDevice string, targetPath strin
 // Returns: is <target mounted, error if occured>
 func (d *NodeService) isTargetMounted(targetPathWithHostPrefix string, isFSVolume bool) (bool, error) {
 	logger.Debugf("Check if target {%s} is mounted", targetPathWithHostPrefix)
+	isMounted, err := d.IsMounted(targetPathWithHostPrefix, isFSVolume)
+	if isMounted && err == nil {
+		logger.Warningf("Idempotent case : targetPath already mounted (%s), so no need to mount again. Finish NodePublishVolume", targetPathWithHostPrefix)
+	}
+	return isMounted, err
+}
+
+func (d *NodeService) IsMounted(targetPathWithHostPrefix string, isFSVolume bool) (bool, error) {
```

Comment:
do we still need this?

Reply:
I think it's better to separate it that every function will do its own purpose, if you think it's not necessary I can roll it back to his older version

Reply:
let's keep the diff minimal and roll it back for now

Reply:
done